core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1974,6 +1991,7 @@ class SparkContext(config: SparkConf) extends Logging {
       }
       _heartbeater = null
     }
+    _shuffleDriverComponents.cleanupApplication()
```

Comment:
Exceptions need to be handled.

Reply:
Ping.

Reply:
I removed IOException from `cleanApplication()`, or are you saying that we should catch runtime exceptions too?

Reply:
> are you saying that we should catch runtime exceptions too?

Yes, otherwise the context will not be completely stopped.